code/go/0chain.net/smartcontract/common/wallet.go

```diff
@@ -0,0 +1,41 @@
+package common
+
+import (
+	"encoding/hex"
+
+	"0chain.net/chaincore/config"
+	"0chain.net/core/common"
+	"0chain.net/core/encryption"
+	"github.com/0chain/common/core/logging"
+)
+
+
+func ValidateWallet(publicKey, delegateWalletID string) *common.Error {
+	if config.Development() {
+		return nil
+	}
+
+	operationalWalletID, err := GetWalletIdFromPublicKey(publicKey)
+	if err != nil {
+		return common.NewError("add_sharder",
+			"couldn't decode publick key to compare to delegate wallet")
+	}
+
+	if operationalWalletID == delegateWalletID {
+		logging.Logger.Error("couldn't use the same wallet as both operational and delegate")
+		return common.NewError("add_sharder",
+			"couldn't use the same wallet as both operational and delegate")
+	}
+
+	return nil
+}
+
+/*GetWalletIdFromPublicKey - given the PK of the provider, return the its operational walletID i.e. the key used to sign the txns */
+func GetWalletIdFromPublicKey(pk string) (string, error) {
```

Comment:
Let's move this function into `encryption` package, and name it as `GetClientIDFromPublickKey()`, comment with `//` instead of `/**/`.